salt/transport/zeromq.py

```diff
@@ -736,6 +736,17 @@ def publish(self, load):
         if load['tgt_type'] == 'list':
             int_payload['topic_lst'] = load['tgt']
 
+        # If zmq_filtering is enabled, target matching has to happen master side
+        if self.opts['zmq_filtering']:
```

Comment:
Personally, I think that dropping the other types for the time being is reasonable. Obviously, we'd want to ensure this is well-documented so we don't trip people up.